### PR TITLE
Remove "es6-symbol" package from DevTools

### PR DESCRIPTION
**Pull request by bvaughn**
This polyfill was copied over from DevTools v3 but given the [current browser compatibility for `Symbol`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol#Browser_compatibility), I think it's safe to remove.

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/Components/HooksTree.js [hunk 1]
```diff
@@ -124,7 +124,7 @@ function HookView({canEditHooks, hook, id, inspectPath, path}: HookViewProps) {
         hook !== null &&
         typeof hook === 'object' &&
         hook.hasOwnProperty(meta.type)
-          ? hook[meta.type]
+          ? hook[(meta.type: any)]
```
Comment (trueadm) [inline packages/react-devtools-shared/src/devtools/views/Components/HooksTree.js]: Why was this change needed?

Comment (bvaughn) [reply packages/react-devtools-shared/src/devtools/views/Components/HooksTree.js]: Flow complains otherwise:
> Cannot access computed property using Symbol

**Comment by bvaughn** [comment]
> Thanks so much for cleaning this up! Do you mind deploying a new version of the devtools into xplat/js after merging this?

Landing things in xplat can be a hassle though, and this isn't super high priority for me at the moment. I'll take a shot at it though, assuming it doens't take *too* long.
